zanata-war/src/main/java/org/zanata/rest/service/SourceDocResourceService.java

```diff
@@ -421,7 +421,7 @@ private HLocale validateTargetLocale(LocaleId locale, String projectSlug,
         } catch (ZanataServiceException e) {
             log.warn(
                     "Exception validating target locale {} in proj {} iter {}",
-                    e, locale, projectSlug, iterationSlug);
+                    locale, projectSlug, iterationSlug);
```

Comment:
Actually, this can work, to fix it we just need to move the exception to
the end of the varargs list, and slf4j will include the stack trace as
intended.

Or, if we just want the exception's `toString()`, we can add another `{}`
to the template.  But as it's only WARN, I would be inclined to keep the
stack trace.

On 20 March 2015 at 11:37, Patrick Huang <notifications@github.com> wrote:

> In
> zanata-war/src/main/java/org/zanata/rest/service/SourceDocResourceService.java
> <https://github.com/zanata/zanata-server/pull/642#discussion_r26813442>:
>
> > @@ -421,7 +421,7 @@ private HLocale validateTargetLocale(LocaleId locale, String projectSlug,
> >          } catch (ZanataServiceException e) {
> >              log.warn(
> >                      "Exception validating target locale {} in proj {} iter {}",
> > -                    e, locale, projectSlug, iterationSlug);
> > +                    locale, projectSlug, iterationSlug);
>
> This is simply wrong. You can't mix template with exception stacktrace.
> This was treating the exception as first variable and log its toString as
> locale.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/642/files#r26813442>.
>